File: 2028-find-missing-observations/2028-find-missing-observations.cpp
class Solution {
public:
    vector<int> missingRolls(vector<int>& rolls, int mean, int n) {
        int sum = accumulate(rolls.begin(), rolls.end(), 0);
        int miss = mean * (n + rolls.size()) - sum;
        if (miss < n || miss > 6 * n) {
            return {};
        }
        int extra = miss % n;
        vector<int> res(n, miss / n);
        for (int i = 0; i < extra; ++i) {
            res[i]++;
        }
        return res;
    }
};

File: 2028-find-missing-observations/2028-find-missing-observations.go
func missingRolls(rolls []int, mean int, n int) []int {
	sum := 0
	for _, num := range rolls {
		sum += num
	}
	miss := mean*(n+len(rolls)) - sum
	if miss < n || miss > 6*n {
		return make([]int, 0)
	}
	extra := miss % n
	val := miss / n
	res := make([]int, n)
	for i := 0; i < n; i++ {
		res[i] = val
		if i < extra {
			res[i]++
		}
	}
	return res
}